pkg/utils/cadvisor_linux.go

```diff
@@ -0,0 +1,236 @@
+package utils
+
+import (
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	cmemory "github.com/google/cadvisor/cache/memory"
+	cadvisorcontainer "github.com/google/cadvisor/container"
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+	cmanager "github.com/google/cadvisor/manager"
+	csysfs "github.com/google/cadvisor/utils/sysfs"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	statsapi "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+)
+
+var (
+	CManager     cmanager.Manager
+	cManagerOnce sync.Once
+)
+
+type CadvisorProvider struct {
+	Manager   cmanager.Manager
+	podLister corelisters.PodLister
+}
+
+func NewCadvisorManager() (cmanager.Manager, error) {
+	cManagerOnce.Do(func() {
+		klog.Infof("CManager is nil")
+		var includedMetrics = cadvisorcontainer.MetricSet{
+			cadvisorcontainer.CpuUsageMetrics:         struct{}{},
+			cadvisorcontainer.ProcessSchedulerMetrics: struct{}{},
+		}
+
+		var allowDynamic bool = true
+		var maxHousekeepingInterval time.Duration = 10 * time.Second
+		var memCache = cmemory.New(10*time.Minute, nil)
+		var sysfs = csysfs.NewRealSysFs()
+		var maxHousekeepingConfig = cmanager.HouskeepingConfig{Interval: &maxHousekeepingInterval, AllowDynamic: &allowDynamic}
+
+		m, err := cmanager.New(memCache, sysfs, maxHousekeepingConfig, includedMetrics, http.DefaultClient, []string{CgroupKubePods}, "")
+		if err != nil {
+			klog.Errorf("Failed to create cadvisor manager start: %v", err.Error())
+		}
+		if err := m.Start(); err != nil {
+			klog.Errorf("Failed to start cadvisor manager: %v", err)
+		}
+		CManager = m
+	})
+	return CManager, nil
+}
+
+func NewCadvisorProvider(manager cmanager.Manager, podLister corelisters.PodLister) *CadvisorProvider {
+	return &CadvisorProvider{
+		Manager:   manager,
+		podLister: podLister,
+	}
+}
+
+func (c *CadvisorProvider) GetCPUAndMemoryStats() (*statsapi.Summary, error) {
```

Comment:
I think all these logic should be merged into collector module, will be more in line with the design principles of each module.

Reply:
This func is no need? 
cpuUsageSample, schedRunqueueTime := caculateCPUUsage(&v, &state) in cadvisor_linux.go calculate the cpu usage, which is just the usage diff/time diff, we can directly use it. The name container_cpu_total_usage is really misleading.